consensus/src/experimental/buffer_manager.rs

```diff
@@ -17,38 +35,133 @@ pub struct ResetRequest {
 pub struct OrderedBlocks {
     pub blocks: Vec<ExecutedBlock>,
     pub finality_proof: LedgerInfoWithSignatures,
+    pub callback: StateComputerCommitCallBackType,
 }
 
+/*
+ *                       LedgerInfo Buffer
+ *
+ *                      ┌────────────┐                    ┌────────────┐
+ *                      │ LedgerInfo │                    │ LedgerInfo │
+ *                  ────► BufferItem ├────────────────────► BufferItem ├───►
+ *                      └─────┬──────┘                    └─────┬──────┘
+ *                            │ Link                            │ Link
+ *                            │                                 │
+ *     ┌────────────┐   ┌─────▼──────┐   ┌────────────┐   ┌─────▼──────┐
+ *     │ BufferItem │   │ BufferItem │   │ BufferItem │   │ BufferItem │
+ * ────►            ├───►            ├───►            ├───►            ├───►
```

Comment:
I don't fully understand this structure, but I guess we'll see in follow up pr

Reply:
yeah, we maintain a two-layer linked list. The LedgerInfo layer is for faster signature aggregation and exchanging with the signing phase (so that the buffer manager does not have to travel through blocks between ledger infos).

Reply:
similar to a skip list